### PR TITLE
Fix psalm errors

### PR DESCRIPTION
**Pull request by alcaeus**
This fixes a couple of errors that appeared due to a psalm update which updated the PHPC method signatures.

### CODE REVIEW

File: psalm-baseline.xml [hunk 1]
```diff
@@ -114,10 +114,16 @@
...
+    <LessSpecificReturnStatement>
+      <code><![CDATA[(object) $this->getArrayCopy()]]></code>
+    </LessSpecificReturnStatement>
     <MixedAssignment>
       <code>$this[$key]</code>
       <code>$value</code>
     </MixedAssignment>
+    <MoreSpecificReturnType>
+      <code>stdClass</code>
+    </MoreSpecificReturnType>
```
Comment (alcaeus) [inline psalm-baseline.xml:126]: The `LessSpecificReturnStatement` and `MoreSpecificReturnType` are due to https://github.com/vimeo/psalm/issues/8187. In essence, Psalm takes the return value of `(object) []` to be `object`, even though it is `stdClass`.

File: psalm-baseline.xml [hunk 2]
```diff
@@ -509,6 +515,9 @@
     </MixedAssignment>
   </file>
   <file src="src/Operation/FindOneAndReplace.php">
+    <MixedArgument>
+      <code><![CDATA[$options['codec'] ?? null]]></code>
+    </MixedArgument>
```
Comment (alcaeus) [inline psalm-baseline.xml:520]: Not sure why this errors in that one specific file - it works in all other files where we have the same checks on the codec option and then use it in a typed context 🤷

**Review by GromNaN — APPROVED** [review]
Perhaps we should set a stronger constraint on the psalm version, since there is no `compose.lock`.

**Comment by alcaeus** [comment]
> Perhaps we should set a stronger constraint on the psalm version, since there is no `compose.lock`.

Yes, perhaps paired with dependabot updating the version constraint range when an update was released. I'll give it some thought.

File: psalm-baseline.xml (hunk 2 above)
Comment (GromNaN) [reply psalm-baseline.xml:520]: I couldn't reproduce.
https://psalm.dev/r/609a35bce9

**Pull request merged**